mmv1/templates/terraform/post_create/spanner_database.go.erb

```diff
@@ -72,4 +72,34 @@ if retentionPeriodOk || ddlOk {
 			return fmt.Errorf("Error waiting to run DDL against newly-created Database: %s", err)
 		}
 	}
+}
+
+enableDropProtection, enableDropProtectionOk := d.GetOk("enable_drop_protection")
+dropProtection := enableDropProtection.(bool)
+if enableDropProtectionOk && dropProtection {
+	updateMask := []string{"enableDropProtection"}
+	url, err := tpgresource.ReplaceVars(d, config, "{{SpannerBasePath}}projects/{{project}}/instances/{{instance}}/databases/{{name}}")
+	if err != nil {
+		return err
+	}
+	// updateMask is a URL parameter but not present in the schema, so ReplaceVars
+	// won't set it
+	url, err = transport_tpg.AddQueryParams(url, map[string]string{"updateMask": strings.Join(updateMask, ",")})
+	if err != nil {
+		return err
+	}
+	res, err = transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+		Config: config,
+		Method: "PATCH",
+		Project: billingProject,
+		RawURL: url,
+		UserAgent: userAgent,
+		Body: obj,
+		Timeout: d.Timeout(schema.TimeoutUpdate),
+	})
+	if err != nil {
+		return fmt.Errorf("Error updating enableDropDatabaseProtection on Database: %s", err)
+	} else {
+		log.Printf("[DEBUG] Finished updating enableDropDatabaseProtection %q: %#v", d.Id(), res)
+	}
```

Comment:
qq: why do we need it for creation?

Reply:
The create database API does not support setting enableDropProtection, so we need ot make seperate request using this https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances.databases/patch